### DISCUSSION TITLE
rabbitmqctl.bat export_definitions does not include all system exchanges (plugin's exchanges are missing)

### DISCUSSION THREAD

**Discussion by iqesak** [rabbitmq/rabbitmq-server#2422] [Other] [answered]
* RabbitMQ version 3.8.2
* Erlang version 21.3
* Operating system, version: Windows 10 Enterprise 64bit, 16GB RAM,  
* enabled_plugins [rabbitmq_auth_mechanism_ssl,rabbitmq_lvc_exchange,rabbitmq_management,rabbitmq_stomp,rabbitmq_tracing,rabbitmq_web_stomp].

"rabbitmqctl.bat export_definitions" command does not include "amq.rabbitmq.trace"  Exchange added by "rabbitmq_tracing" plugin, even if it includes all system Exchanges like: "amq.direct", "amq.fanouts", ....

When I use command "rabbitmqctl,bat list_exchanges", I can see all Exchages correctly including "amq.rabbitmq.trace" and all system exchanges.

It would be good feature beside fixing this, to have an option to exclude all system Exchanges from export, since they are anyway ignored during import.

kind regards
Milos

**Comment by michaelklishin** [comment, marked as answer]
`amq.rabbitmq.trace` is not included because it is managed by virtual host tracing. `rabbitmq_tracing` simply exposes the results of tracing in a text file accessible via management UI. There are no plans to change this. There are no plans to exclude the built-in exchanges either: that would cause more questions and confusion that including them today does.
